### DISCUSSION TITLE
[Low Priority Request] AFK Self-Kick on Timer

### DISCUSSION THREAD

**Discussion by mercurialmusic** [Zrips/CMI#4374] [Feature Requests]
### Description of feature/improvement request: ###
Making this request on behalf of some of my players who have asked for it. Potentially related to #4158. 

Mostly for those who have the afk kick bypass already, this would be an additional argument to the /afk command itself that would kick a player after a specified amount of time. For example if they want to afk for an hour but don't want to be online all night, they would use, for example, `/afk 60m`, then the server would kick them after 60 minutes. 

This would need a configurable maximum amount of time that players wouldn't be able to exceed.

Please ignore this request if there would be any increase in resources needed for these sorts of checks. We were just curious if the current afk system would be able to manage that sort of thing pretty much as is, just with a variable timer.

---
### Basic explanation as to why this feature is needed and/or useful to have: ###
Useful for those with the afk bypass who don't want to be resource hogs but who are incapable of setting their own timers I guess? The auto-kick bypass is a great perk to be able to give, but there are those who either abuse it or who forget they had the game running at all (I've definitely been there before). Adding this might help to minimize the accidental afk sessions at least.

**Comment by mrfloris** [comment]
You can go to alias.yml and set /afk to false, and make a new afk command using cmi aliaseditor.
Then use the perm:! specialized command to check against the bypass mode, and check:! to see if the $1 variable is set to expected integer. 

Then when they use /afk something, it will work as normal, but /afk 50 would try to set a delay! $1 after which it would kick.

The specifics of this just requires some testing, but you can already achieve this using existing cmi features.

**Comment by mercurialmusic** [comment]
> You can go to alias.yml and set /afk to false, and make a new afk command using cmi aliaseditor.
> Then use the perm:! specialized command to check against the bypass mode, and check:! to see if the $1 variable is set to expected integer.
> 
> Then when they use /afk something, it will work as normal, but /afk 50 would try to set a delay! $1 after which it would kick.
> 
> The specifics of this just requires some testing, but you can already achieve this using existing cmi features.

Brilliant, thanks! Will test.
